pandas/core/indexes/datetimes.py

```diff
@@ -1364,7 +1378,7 @@ def _parsed_string_to_bounds(self, reso, parsed):
             st = datetime(parsed.year, parsed.month, parsed.day,
                           parsed.hour, parsed.minute, parsed.second,
                           parsed.microsecond)
-            return (Timestamp(st, tz=self.tz), Timestamp(st, tz=self.tz))
+            return Timestamp(st, tz=self.tz), Timestamp(st, tz=self.tz)
```

Comment:
Since you're already making edits here there's a small bug-like that might be worth fixing.  The day, hour, minute, and second cases don't have `tz=self.tz` passed to the upper half of the returned tuple.

Reply:
Thanks for pointing that out! By now the timezone is like ignored (or at least not considered well enough) in the later stages but maybe one day that will change.